### ISSUE TITLE
Copy/Cut/Paste does not work when remapping shortcuts on OSX

### ISSUE THREAD

**Issue by jdthorpe** [codemirror/CodeMirror#5848]
This is a niche issue to be sure, but I thought I'd report it anyway:  On OSX, if you re-map the default keyboard shortcuts for cut/copy/paste (i.e. <kbd>⌘</kbd><kbd>x</kbd>, <kbd>⌘</kbd><kbd>c</kbd>, <kbd>⌘</kbd><kbd>v</kbd>) to custom keyboard shortcuts such as (<kbd>^</kbd><kbd>x</kbd>, <kbd>6</kbd><kbd>c</kbd>, <kbd>^</kbd><kbd>v</kbd>) which emulate the Windows keyboard experience for those of us who use both Win and OSX on an hourly basis.   

The issue is that with these shortcuts in place, neither the default nor custom nor default keyboard shortcuts work in codemirror.  The only way around that I've found is to re-map the default keyboard shortcuts for browsers, which makes copy/pasting across all browser interfaces a pain.   As a coder, at least a quarter of my online typing experience is in codemirror, so it's essential that I have theses shortcuts working.  On the other hand reverting these shortcuts for browsers *just to accommodate code mirror* makes my entire online experience worse

By way of background, I remapped the the shortcuts like so: 
![Screen Shot 2019-04-10 at 10 38 14 AM](https://user-images.githubusercontent.com/10930181/55903492-a1cdb580-5b82-11e9-9e42-eb220d862c7e.png)

and re-maped them to chrome like so: 

![Screen Shot 2019-04-10 at 10 41 44 AM](https://user-images.githubusercontent.com/10930181/55903493-a2664c00-5b82-11e9-9718-ea4cf9b73e96.png)

**Comment by marijnh** [comment]
That's odd—the editor listens for the `"copy"`/`"cut"`/`"paste"` events, which I'd expect to be fired regardless of keyboard layout. There's no hard-coded references to specific clipboard-related keys in the codebase at all.

**Comment by jdthorpe** [comment]
Yep. It is odd.  I'll keep fiddling, but the behavior is pretty reliable on chrome on my machine.  I'll keep you posted if I manage to get it working. It's only been acting like this for a few weeks, so maybe it's a recent chrome bug.

**Comment by jdthorpe** [comment]
I'm pretty sure [this line](https://github.com/codemirror/CodeMirror/blob/ed8dfeb5e2ed25b5dd1f1eccc7b757ca6dbd118d/src/input/keymap.js#L33) is the culpret.  I noticed that the remapped Cut (`ctrl-x`) and Copy (`ctrl-c`) work fine but my map for paste does not work (`ctrl-v`).  Specifically,  the edit menu does not flash when pressing `ctrl-v` to indicate that the paste action is being emitted.  Furthermore when I add more lines than fit in a Codemirror instance, `ctrl-v` is indeed mapped to page-down.  

It would be awesome if the there was a code-mirror Chrome extension to allow setting global defaults (across web sites, that is),  but in know that's a huge ask, so I'm not expecting a quick fix...

**Comment by marijnh** [comment]
> I'm pretty sure this line is the culpret.

Ah, good catch, yes, we're emulating the macOS default behavior for ctrl-v, and that will indeed block your paste shortcut. Unfortunately, the browser doesn't provide any reasonable way to figure out what key bindings it is using.

You could try to write an extension that does something like `let cm =  document.querySelector(".CodeMirror"); if (cm) cm.CodeMirror.constructor.keyMap.emacsy["Ctrl-Y"]`—that should kill that keybinding if there's an active editor in the page.

**Comment by jdthorpe** [comment]
So, I went a little overboard and created [a Chrome extension](https://github.com/jdthorpe/codemirror-global-settings) which, in addition to the above two lines, will disable user selected bindings and set the theme and key bindings for every CodeMirror site on the web.  Due to the permissions it requires, it has to go through an extended review, so not sure when it will be up on the Chrome Extension store, if at all.
